group-coordinator/src/main/java/org/apache/kafka/coordinator/group/metrics/CoordinatorRuntimeMetrics.java

```diff
@@ -53,6 +53,21 @@ public interface CoordinatorRuntimeMetrics extends AutoCloseable {
      */
     void recordEventQueueProcessingTime(long durationMs);
 
+    /**
+     * Record the event purgatory time.
+     *
+     * @param purgatoryTimeMs    The time the event was completed.
+     * @param writeTimestamp     The time the records were written to the log.
+     */
+    void recordEventPurgatoryTime(long purgatoryTimeMs, long writeTimestamp);
```

Comment:
nit: Should `purgatoryTimeMs` be called `durationMs` like the others? Moreover, I don't really understand why we need `writeTimestamp` here. It does not seem necessary.